test-data/unit/check-inference.test

```diff
@@ -840,7 +828,7 @@ for x in [A()]:
     b = x # E: Incompatible types in assignment (expression has type "A", variable has type "B")
     a = x
 
-for y in []: # E: Need type annotation for variable
+for y in []:
```

Comment:
What's the type of `y` here? I think that the original message was fine. At least add a `reveal_type`.